src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/on_ui_language_change_notification.cc

```diff
@@ -80,27 +80,37 @@ void OnUILanguageChangeNotification::Run() {
   (*message_)[strings::params][strings::function_id] =
       static_cast<int32_t>(mobile_apis::FunctionID::OnLanguageChangeID);
 
-  const ApplicationSet& accessor =
-      application_manager_.applications().GetData();
-
-  ApplicationSetConstIt it = accessor.begin();
-  for (; accessor.end() != it;) {
-    ApplicationSharedPtr app = *it;
-    ++it;
-    (*message_)[strings::params][strings::connection_key] = app->app_id();
+  std::vector<uint32_t> app_list;
+  std::vector<uint32_t> to_unregister;
+
+  {
+    const ApplicationSet& accessor =
+        application_manager_.applications().GetData();
+    auto message_language =
+        (*message_)[strings::msg_params][strings::hmi_display_language].asInt();
+
+    for (auto app : accessor) {
+      auto app_id = app->app_id();
+      app_list.push_back(app_id);
+      if (app->ui_language() != message_language) {
+        to_unregister.push_back(app_id);
+      }
+    }
+  }
+
+  for (auto app_id : app_list) {
+    (*message_)[strings::params][strings::connection_key] = app_id;
     SendNotificationToMobile(message_);
```

Comment:
Any reason that we can't send this during the initial loop?

Reply:
Regardless, it seems like we could just create a copy of the app list rather than pushing each app ID individually if this separation is still necessary

Reply:
Good idea, this solution does look cleaner! I have changed from using a vector of appIDs to copying the full ApplicationSet in https://github.com/smartdevicelink/sdl_core/pull/3806/commits/e3d70c4092dc2b5d38f9aebb47bfeab00ac21a6c